source/grid/tria.cc

```diff
@@ -10190,6 +10190,16 @@ Triangulation<dim, spacedim>::clear()
 }
 
 
+template <int dim, int spacedim>
+const MPI_Comm &
+Triangulation<dim, spacedim>::get_communicator() const
+{
+  static const MPI_Comm comm = MPI_COMM_SELF;
+
+  return comm;
+}
```

Comment:
@bangerth Would this be fine? Or do you prefer that we remove the reference?

Reply:
That works as well. `MPI_Comm` objects are copyable (they must be -- they need to work in C, in which all data types are copyable), but we typically pass them by reference. So this is fine with me.

Reply:
Given that we do not store a communicator, I would lean towards returning `MPI_Comm` by value. (Background: I think we should keep global static variables low and only do if necessary. The global startup part of deal.II called before main is very big already.)
However, this technically breaks the backwards compatibility of the derived parallel classes, which used to take a reference. Given the typical use, I would still vote for returning by value, but I would be fine either way.

Reply:
I have changed it so that it return by value. BTW: my first implementation (pre-PR version) had a dummy communicator inside of `Triangulation`.